### DISCUSSION TITLE
SQLServer db pull was empty

### DISCUSSION THREAD

**Discussion by AugustoGCP** [prisma/prisma#19201] [Q&A]
Hey guys!

I have a SQL Server DB in production and I would like to create a dashboard using JS and Prisma ORM. When I run ```npm prisma db pull``` the folling error is showed:

```
Prisma schema loaded from prisma\schema.prisma
Environment variables loaded from .env
Datasource "db": SQL Server database

× Introspecting based on datasource defined in prisma\schema.prisma
Error:
P4001 The introspected database was empty:

prisma db pull could not create any models in your schema.prisma file and you will not be able to generate Prisma Client with the prisma generate command.

To fix this, you have two options:

- manually create a table in your database.
- make sure the database connection URL inside the datasource block in schema.prisma points to a database that is not empty (it must contain at least one table).

Then you can run prisma db pull again.
```
Those are the Prisma files:

prisma.schema
```
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "sqlserver"
  url      = env("DATABASE_URL")
}
```

.env
```
# Environment variables declared in this file are automatically made available to Prisma.
# See the documentation for more detail: https://pris.ly/d/prisma-schema#accessing-environment-variables-from-the-schema

# Prisma supports the native connection string format for PostgreSQL, MySQL, SQLite, SQL Server, MongoDB and CockroachDB.
# See the documentation for all the connection string options: https://pris.ly/d/connection-strings

DATABASE_URL="sqlserver://localhost:1433;database=engemandb;user=USER;password=PASSWORD;encrypt=DANGER_PLAINTEXT"
```

This is the picture of the DataBase
![image](https://github.com/prisma/prisma/assets/49987065/8fe205ac-1afc-4a80-920d-980283540710)

I dont know if I forgot something or I'm doing something wrong. Could anybody help me?

**Comment by nurul3101** [comment]
Hey @AugustoGCP 👋 
Are you using the latest version of Prisma i.e. `4.14.0`?

If you try connecting in a database IDE with this connection string - `sqlserver://localhost:1433;database=engemandb;user=USER;password=PASSWORD;encrypt=DANGER_PLAINTEXT`

Are you able to see all the tables?

**Reply by AugustoGCP** [reply]
Could you explain to me how to do this test of connection please?

Follow the Environment

```
prisma                  : 4.14.0
@prisma/client          : Not found
Current platform        : windows
Query Engine (Node-API) : libquery-engine d9a4c5988f480fa576d43970d5a23641aa77bc9c (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Migration Engine        : migration-engine-cli d9a4c5988f480fa576d43970d5a23641aa77bc9c (at node_modules\@prisma\engines\migration-engine-windows.exe)
Format Wasm             : @prisma/prisma-fmt-wasm 4.14.0-67.d9a4c5988f480fa576d43970d5a23641aa77bc9c
Default Engines Hash    : d9a4c5988f480fa576d43970d5a23641aa77bc9c
Studio                  : 0.484.0
```

**Reply by nurul3101** [reply]
What I meant was that can you try connecting to the database with the mentioned connection string through a database IDE like DataGrip or TablePlus, or any other database GUI?

**Reply by nurul3101** [reply]
@AugustoGCP Did you had a chance to look at my last response?

If it answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
We are closing this Discussion due to inactivity. Please open a new Discussion if you have a similar problem.
